src/libs/l2ps/zk/L2PSBatchProver.ts

```
@@ -0,0 +1,584 @@
+/**
+ * L2PS Batch Prover
+ * 
+ * Generates PLONK proofs for L2PS transaction batches.
+ * Automatically selects the appropriate circuit size (5, 10, or 20 tx).
+ * Pads unused slots with zero-amount transfers.
+ */
+
+// Bun compatibility: patch web-worker before importing snarkjs
+const isBun = (globalThis as any).Bun !== undefined;
```

ESLint

[error] Extra semicolon.

(semi)